src/Simulation/Simulators/CommonNativeSimulator/CommonNativeSimulator.cs

```diff
@@ -35,6 +35,8 @@ public abstract partial class CommonNativeSimulator : SimulatorBase, IQSharpCore
 
         public uint Id { get; protected set; }
 
+        public abstract uint[] GetQubitIds();
```

Comment:
Does this need to be specifically a `uint[]`, or can it be an `IEnumerable<uint>`? May also be worth adding an API documentation comment to indicate what's returned by this method.

Reply:
As mentioned below, it may also be reasonable to make the abstract class define a property directly rather than a method:

```suggestion
        public abstract uint[] QubitIds { get; }
```

Reply:
Thank you for helping me to master the abstract properties (I'm still new to C# and I'm still thinking in C++). Updating.
As for `uint[]` vs. `IEnumerable<uint>`, I would prefer to postpone the decision until later.